quinn-proto/src/connection/streams.rs

```diff
@@ -344,16 +689,9 @@ impl Streams {
         None
     }
 
-    /// Access a receive stream due to a message from the peer
-    ///
-    /// Similar to `recv_mut`, but with additional sanity-checks are performed to detect peer
-    /// misbehavior.
-    pub fn recv_stream(
-        &mut self,
-        side: Side,
-        id: StreamId,
-    ) -> Result<Option<&mut Recv>, TransportError> {
-        if side == id.initiator() {
+    /// Check stream `id` received from peer to detect a peer's misbehavior
+    fn validate_received_id(&mut self, id: StreamId) -> Result<(), TransportError> {
```

Comment:
```suggestion
    /// Check for errors entailed by the peer's use of `id` as a send stream
    fn validate_receive_id(&mut self, id: StreamId) -> Result<(), TransportError> {
```
These checks don't apply to *all* stream IDs received from the peer, only the ones that we are intended to associate with what is locally considered a receive stream.